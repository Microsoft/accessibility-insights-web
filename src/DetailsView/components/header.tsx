// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { GearOptionsButtonComponent } from '../../common/components/gear-options-button-component';
import { DropdownClickHandler } from '../../common/dropdown-click-handler';
import { FeatureFlags } from '../../common/feature-flags';
import { DetailsViewPivotType } from '../../common/types/details-view-pivot-type';
import { FeatureFlagStoreData } from '../../common/types/store-data/feature-flag-store-data';
import { title } from '../../content/strings/application';
import { Switcher, SwitcherDeps } from '../components/switcher';

export type HeaderDeps = SwitcherDeps;

export interface IHeaderProps {
    deps: HeaderDeps;
    dropdownClickHandler: DropdownClickHandler;
    featureFlagStoreData: FeatureFlagStoreData;
    tabClosed: boolean;
    selectedPivot: DetailsViewPivotType;
}

export class Header extends React.Component<IHeaderProps> {
    public render(): JSX.Element {
        return (
            <header className="header-bar">
                <img className="header-icon" src="../../icons/brand/white/brand-white-48px.png" alt="" />
                <div className="ms-font-m header-text">{title}</div>
                {this.renderSwitcher()}
                {this.renderButton()}
            </header>
        );
    }

    private renderSwitcher(): JSX.Element {
        if (!this.shouldRenderSwitcher()) {
            return null;
        }

        return <Switcher deps={this.props.deps} pivotKey={this.props.selectedPivot} />;
    }

    private shouldRenderSwitcher(): boolean {
        const { featureFlagStoreData, tabClosed } = this.props;

        return featureFlagStoreData != null && featureFlagStoreData[FeatureFlags.newAssessmentExperience] === true && tabClosed === false;
    }

    private renderButton(): JSX.Element {
        if (this.props.tabClosed) {
            return null;
        }

        return (
            <GearOptionsButtonComponent
                dropdownClickHandler={this.props.dropdownClickHandler}
                featureFlags={this.props.featureFlagStoreData}
            />
        );
    }
}
