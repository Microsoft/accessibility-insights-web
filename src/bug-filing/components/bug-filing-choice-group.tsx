// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ChoiceGroup, IChoiceGroupOption } from 'office-ui-fabric-react/lib/ChoiceGroup';
import * as React from 'react';

import { UserConfigMessageCreator } from '../../common/message-creators/user-config-message-creator';
import { NamedSFC } from '../../common/react/named-sfc';
import { BugFilingService } from '../types/bug-filing-service';

export type BugFilingChoiceGroupProps = {
    deps: BugFilingChoiceGroupDeps;
    selectedBugFilingService: BugFilingService;
    bugFilingServices: BugFilingService[];
};

export type BugFilingChoiceGroupDeps = {
    userConfigMessageCreator: UserConfigMessageCreator;
};

export const BugFilingChoiceGroup = NamedSFC<BugFilingChoiceGroupProps>('BugFilingChoiceGroup', props => {
    const { userConfigMessageCreator } = props.deps;

    const getOptions: () => IChoiceGroupOption[] = () => {
        return props.bugFilingServices.map(service => {
            return {
                key: service.key,
                text: service.displayName,
            };
        });
    };

    const onChange = (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, option?: IChoiceGroupOption) => {
        userConfigMessageCreator.setBugService(option.key);
    };

    return (
        <ChoiceGroup
            className={'bug-filing-choice-group'}
            onChange={onChange}
            options={getOptions()}
            selectedKey={props.selectedBugFilingService.key}
        />
    );
});
