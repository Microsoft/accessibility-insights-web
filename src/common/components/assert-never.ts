// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

// This function helps with exhaustiveness checking in switch statements
// If used in the default case, TS will throw an error unless all cases are handled.
export const assertNever = (x: never): never => {
    throw Error(
        'This error should never be thrown when strict null checks are applied to this project',
    );
};
