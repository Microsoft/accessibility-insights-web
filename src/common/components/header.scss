// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';
@import '../../common/styles/common.scss';
@import '../../common/styles/fonts.scss';

.header-bar {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: nowrap;

    background-color: $ada-brand-color;
    height: $detailsViewHeaderBarHeight;
    width: 100%;

    > * {
        flex-shrink: 0;
    }

    :global(.header-icon) {
        margin-left: 16px;
        height: 22px;
    }

    .header-text {
        // Note: this is slightly off from the mock-up (which says 40px); we're using 32px
        // temporarily until we add more complete reflow behavior for the header bar because
        // 32px is just small enough for a user at default zoom levels to resize the window
        // to the minimum 500px allowed by Chrome and not have the title get cut off.
        margin-right: 32px;
        margin-left: 8px;

        color: $header-bar-title-color;
        font-size: $fontSizeML;
        font-family: $fontFamily;

        // This is for the benefit of higher-zoom-level users.
        flex-shrink: 1;
        @include ellipsedText();
    }
}
