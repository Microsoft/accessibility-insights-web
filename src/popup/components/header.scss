// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

.launch-panel-header {
    margin-bottom: 12px;
    margin-left: 14px;
    margin-right: 14px;

    padding: 0 8px;

    .header-subtitle {
        color: $neutral-60;
    }
}
