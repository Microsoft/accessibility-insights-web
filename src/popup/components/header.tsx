// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { css } from '@uifabric/utilities';
import * as React from 'react';
import { NamedFC } from '../../common/react/named-fc';

export interface HeaderProps {
    title: string;
    subtitle?: React.ReactChild;
    rowExtraClassName?: string;
    children?: JSX.Element;
}

export const Header = NamedFC<HeaderProps>('Header', props => {
    return (
        <header className="ms-Grid launch-panel-header">
            <div className={css('ms-Grid-row', props.rowExtraClassName)}>
                <div
                    role="heading"
                    aria-level={1}
                    className="ms-Grid-col ms-u-sm10 ms-fontColor-neutralPrimary ms-font-xl old"
                >
                    {props.title}
                </div>
                {props.children}
            </div>
            <div className="header-subtitle ms-fontWeight-semilight ms-fontSize-xs">
                {props.subtitle}
            </div>
        </header>
    );
});
