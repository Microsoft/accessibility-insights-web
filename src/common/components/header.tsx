// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { HeaderIcon, HeaderIconDeps } from 'common/components/header-icon';
import { NamedFC } from 'common/react/named-fc';
import { title } from 'content/strings/application';
import * as styles from './header.scss';

export type HeaderProps = { deps: HeaderDeps };
export type HeaderDeps = HeaderIconDeps;

export const Header = NamedFC<HeaderProps>('Header', props => {
    return (
        <header className={styles.headerBar}>
            <HeaderIcon deps={props.deps} />
            <span className={styles.headerText}>{title}</span>
            {props.children}
        </header>
    );
});
