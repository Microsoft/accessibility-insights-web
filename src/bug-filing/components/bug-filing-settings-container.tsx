// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../common/react/named-sfc';
import { BugServiceProperties } from '../../common/types/store-data/user-configuration-store';
import { SettingsDeps } from '../../DetailsView/components/settings-panel/settings/settings-props';
import { BugFilingServiceProvider } from '../bug-filing-service-provider';
import { BugFilingService } from '../types/bug-filing-service';
import { BugFilingChoiceGroup, BugFilingChoiceGroupDeps } from './bug-filing-choice-group';

export interface BugFilingSettingsContainerProps {
    deps: BugFilingSettingsContainerDeps;
    selectedBugFilingService: BugFilingService;
    selectedBugFilingServiceData: BugServiceProperties;
}

export type BugFilingSettingsContainerDeps = {
    bugServiceProvider: BugFilingServiceProvider;
} & BugFilingChoiceGroupDeps &
    SettingsDeps;

export const BugFilingSettingsContainer = NamedSFC<BugFilingSettingsContainerProps>('BugFilingSettingsContainer', props => {
    const { deps, selectedBugFilingService, selectedBugFilingServiceData } = props;
    const SettingsForm = selectedBugFilingService.renderSettingsForm;
    const bugFilingServices = deps.bugServiceProvider.allVisible();

    return (
        <>
            <BugFilingChoiceGroup deps={deps} selectedBugFilingService={selectedBugFilingService} bugFilingServices={bugFilingServices} />
            <SettingsForm deps={deps} settings={selectedBugFilingServiceData} />
        </>
    );
});
