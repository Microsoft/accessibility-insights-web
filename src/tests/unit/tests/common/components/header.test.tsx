// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';

import { Header, HeaderDeps } from 'common/components/header';

describe('Header', () => {
    it('renders per snapshot', () => {
        const applicationTitle = 'THE_APPLICATION_TITLE';
        const deps = {
            textContent: {
                applicationTitle,
            },
        } as HeaderDeps;
        const wrapper = shallow(<Header deps={deps} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders without header title', () => {
        const applicationTitle = 'THE_APPLICATION_TITLE';
        const deps = {
            textContent: {
                applicationTitle,
            },
        } as HeaderDeps;
        const wrapper = shallow(<Header deps={deps} showHeaderTitle={false} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it.each([true, false])('renders with showFarItems equals %s', showFarItems => {
        const applicationTitle = 'THE_APPLICATION_TITLE';
        const deps = {
            textContent: {
                applicationTitle,
            },
        } as HeaderDeps;
        const wrapper = shallow(
            <Header
                deps={deps}
                farItems={<div>THis is far items!</div>}
                showFarItems={showFarItems}
            />,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders in narrow mode', () => {
        const applicationTitle = 'THE_APPLICATION_TITLE';
        const deps = {
            textContent: {
                applicationTitle,
            },
        } as HeaderDeps;
        const wrapper = shallow(<Header deps={deps} isNarrowMode={true} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
