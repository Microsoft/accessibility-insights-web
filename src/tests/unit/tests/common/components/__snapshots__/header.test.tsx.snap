// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders per snapshot 1`] = `
<header
  className="headerBar"
>
  <Component
    deps={Object {}}
  />
  <span
    className="headerText"
  >
    Accessibility Insights for Web
  </span>
</header>
`;
